### DISCUSSION TITLE
how to implement a NetLogo-style `diffuse` in Mesa

### DISCUSSION THREAD

**Discussion by NickleDave** [mesa/mesa#1131] [Q&A]
Hi,

I'm working with some NetLogo models that use the command `diffuse`:
http://ccl.northwestern.edu/netlogo/docs/dict/diffuse.html

Are there any existing `mesa` implementations of this command?  
I did a search and didn't find anything.

Now that I actually read their docs, it looks like this is basically a method you execute at the `Model` level that tells a bunch of `Agent`s to update each other:  
> Tells each patch to give equal shares of (number * 100) percent of the value of patch-variable to its eight neighboring patches. number should be between 0 and 1. 

I can definitely try implementing something like that, but thought I'd ask here too.

**Comment by aiwhoo** [comment]
Ironically, alot of mesa users are coming from netlogo. It may be good to have a Netlogo equivalent dictionary for MESA. I am searching for an equivalent import-drawing function.
